### PR TITLE
Restructured Repo to allow for ADO reference (+ minor bugfix)

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Moved logic to central location in utlities to enable reference by both GH & ADO
- Fixed small bug in global.module tests as it turned out there is a misreference which prevented me from testing the ADO pipelines

## Pipeline References
| Name | GH Badge |
| - | - |
| Analysis Services | [AnalysisServices: Servers](https://github.com/MrMCake/ResourceModules/actions/workflows/ms.analysisservices.servers.yml) |
| Virtual Machines | [Compute: Virtualmachines](https://github.com/MrMCake/ResourceModules/actions/workflows/ms.compute.virtualmachines.yml)  |
| Virtual WAN | [Network: Virtualwans](https://github.com/MrMCake/ResourceModules/actions/workflows/ms.network.virtualwans.yml) |

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: utilities/pipelines/resourcePublish/Publish-ModuleToNuget.ps1 [hunk 1]
```diff
@@ -0,0 +1,524 @@
+
```
Comment (eriqua) [inline utilities/pipelines/resourcePublish/Publish-ModuleToNuget.ps1:1]: Did we also add that to the publish step?

File: arm/.global/global.module.tests.ps1 [hunk 2]
```diff
@@ -246,15 +246,15 @@ Describe 'Readme tests' -Tag Readme {
...
+        It '[<moduleFolderName>] parameters section should contain all parameters from the template file' -TestCases $readmeFolderTestCases {
             param(
                 $moduleFolderName,
                 $templateContent,
                 $readMeContent
             )
 
             $ReadmeHTML = ($readMeContent | ConvertFrom-Markdown -ErrorAction SilentlyContinue).Html
-            $parameters = Get-Member -InputObject $templateContent.parameters -MemberType NoteProperty
+            $parameters = $templateContent.parameters.Keys
```
Comment (eriqua) [inline arm/.global/global.module.tests.ps1:257]: Minor: should we align to use either 'Keys' or 'keys' throughout the script?

File: utilities/pipelines/resourcePublish/Publish-ModuleToNuget.ps1 (hunk 1 above)
Comment (MrMCake) [reply utilities/pipelines/resourcePublish/Publish-ModuleToNuget.ps1:1]: Removed. We used it for the Hydration as it was a .psd PowerShell module and we offered the option in the pipeline to publish there too. At least until we merged the ADO pipelines we don't need to keep it

File: arm/.global/global.module.tests.ps1 (hunk 2 above)
Comment (MrMCake) [reply arm/.global/global.module.tests.ps1:257]: Keys it is

**Review by eriqua — DISMISSED** [review]
🆗

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**